pan-domain-auth-example/app/controllers/AdminController.scala

```diff
@@ -1,19 +1,24 @@
 package controllers
 
-import play.api.mvc.{ Action, Controller}
+import javax.inject.Inject
+
+import play.api.Configuration
+import play.api.mvc.{Action, Controller}
 import play.api.Play.configuration
 import play.api.Play.current
+import play.api.libs.ws.WSClient
 
 
-object AdminController extends Controller with ExampleAuthActions {
+class AdminController @Inject() (override val config: Configuration, override val wsClient: WSClient)
```

Comment:
I've heard people arguing that compile time DI is better than run time.

But anyway, this file is just an example...